eng/cmake/global_compile_options.cmake

```diff
@@ -0,0 +1,31 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
```

Comment:
Can we club these cmake stuff under cmake root dir?

Reply:
Re-organized the compile options and modules to both be under a `/cmake` directory. Now we have `/cmake/build` for compile and `/cmake/modules` which is a common practice for modules.